storage/local/chunk.go

```diff
@@ -124,23 +129,27 @@ func (cd *chunkDesc) refCount() int {
 }
 
 func (cd *chunkDesc) firstTime() model.Time {
+	// No lock required, will never be modified.
+	return cd.chunkFirstTime
+}
+
+func (cd *chunkDesc) lastTime() model.Time {
```

Comment:
If we had a lot of concurrent readers, perhaps...

That would indeed be something to be benchmarked.